### PR TITLE
[#156572526] Admin can create API Users

### PR DESCRIPTION
**Pull request by HawiCaesar**
### What does this PR do ?
An admin can create api users
### Description of work done
- add django-oauth-toolkit to requirements
- add django-oauth-toolkit to base.py
- create custom way to add api user to models.py
- add permissions.py to limit to api users only
- write tests for feature
### How can I manually test this ?
- `python manage.py test` OR
- Login into the admin dashboard
- Add 1 or more`Security Users`
- Add an `API User`
- You can simply provide a name for the `API User` without filling any other details
- Click save
- On Postman allow the `API User` you created to access security users via giving it a token
   - As a `POST` request add this as the url `http://127.0.0.1:8000/api/v1/o/token/`
   - In the `Body` section use `form-data` and add the following keys
      - `grant_type` => `client_credentials`
      - `client_secret` => The client secret provided in the `API User` you created
      - `client_id` => The client id provided in the `API User` you created
      - Then click `Send`. You will get back an access_token, expiry time for the token, the token type and the scope of for the token provided
- Then as a `GET` request add this as the url `http://127.0.0.1:8000/api/v1/security-user-emails/`
- In the headers section add `Authorization` and in the value add the word `Bearer ` with a single space. Then add the `access_token` given to you in the `POST` response.
- Click `Send`. You should see a list of Security users.
### Relevant PT Story
[#156572526](https://www.pivotaltracker.com/story/show/156572526)

### Screenshots if appropriate
screen shot 2018-04-11 at 21 15 02

screen shot 2018-04-11 at 21 15 50

screen shot 2018-04-11 at 21 16 22

screen shot 2018-04-11 at 21 16 47

### CODE REVIEW

File: api/urls.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
 
-from .views import UserViewSet, ItemViewSet, SecurityUserViewSet
+from .views import UserViewSet, ItemViewSet, SecurityUserEmailsViewSet
 
 router = SimpleRouter()
 router.register('users', UserViewSet)
 router.register('items', ItemViewSet, 'items')
-router.register('security_users', SecurityUserViewSet, 'security_users')
+router.register('security_user_emails',
+                SecurityUserEmailsViewSet, 'security_user_emails')
```
Comment (egichuri) [inline api/urls.py]: let's use `-` instead of `_` here

Comment (HawiCaesar) [reply api/urls.py]: Okay noted.

Comment (HawiCaesar) [reply api/urls.py]: Done.

File: art/urls.py [hunk 2]
```diff
@@ -27,6 +27,7 @@
     path('jet/', include('jet.urls', 'jet')),
     path('jet/dashboard/', include('jet.dashboard.urls', 'jet-dashboard')),
     path('api-auth/', include('rest_framework.urls')),
+    path('o/', include('oauth2_provider.urls', namespace='oauth2_provider'))
```
Comment (egichuri) [inline art/urls.py]: i'm thinking we move the last 2 urls to `api/urls.py` since they're for the api

Comment (HawiCaesar) [reply art/urls.py]: More for an organization thing yes ?

Comment (egichuri) [reply art/urls.py]: yea.. once it's moved to `api/urls.py`

Comment (egichuri) [reply art/urls.py]: `api-auth/` too

Comment (HawiCaesar) [reply art/urls.py]: Done.

**Pull request merged**